lib/msf/core/exploit/remote/smtp_deliver.rb

```diff
@@ -246,10 +256,16 @@ def smtp_send_recv(cmd, nsock=self.sock)
     begin
       nsock.put(cmd)
       res = nsock.get_once
-      while !(res =~ /(^|\r\n)\d{3}( .*|)\r\n$/) && chunk = nsock.get_once
-        res += chunk
+      while !(res =~ /(^|\r\n)\d{3}( .*|)\r\n$/)
+        chunk = nsock.get_once
+        break unless chunk
+        if res
+          res += chunk
```

Comment:
```suggestion
          res << chunk
```

Local mutation here should be fine - 'for performance'